crypto/asn1/a_time.c

```diff
@@ -627,17 +627,38 @@ time_t ossl_asn1_string_to_time_t(const char *asn1_string)
         ASN1_TIME_free(timestamp_asn1);
         return -1;
     }
+    ASN1_TIME_free(timestamp_asn1);
+
+#if defined(__DJGPP__)
+    /*
+     * This is NOT thread-safe.  Do not use this method for platforms other
+     * than djgpp.
+     */
+    tz = getenv("TZ");
+    if (tz != NULL) {
+        tz = OPENSSL_strdup(tz);
+        if (tz == NULL) {
+            OPENSSL_free(timestamp_tm);
+            return -1;
+        }
+    }
+    setenv("TZ", "UTC", 1);
+
+    timestamp_utc = mktime(timestamp_tm);
 
-#ifdef USE_TIMEGM
+    if (tz != NULL) {
+        setenv("TZ", tz, 1);
+        OPENSSL_free(tz);
+    } else {
+        unsetenv("TZ");
+    }
+#elif defined(USE_TIMEGM)
     timestamp_utc = timegm(timestamp_tm);
-    OPENSSL_free(timestamp_tm);
 #else
-    timestamp_local = mktime(timestamp_tm);
-    OPENSSL_free(timestamp_tm);
-
-    timestamp_utc = timestamp_local - timezone;
+    tzset();
```

Comment:
This causes Windows CI build failure. Is it actually needed? mktime() should call it implicitly. This is outside of DJGPP support so better not add it.

Reply:
It seemed a bit pointless to use a separate variable here and do it in two steps.  Then I realized `timezone` may not even be set until the call to `mktime()`, so doing both in one expression would be undefined behavior.  I added `tzset()` then which I see now doesn't work on Windows.